### ISSUE TITLE
False positive: `${(s..)^:-...}` param-expansion flags misidentified as bash

### ISSUE THREAD

**Issue by GrzegorzKozub** [z-shell/zsh-lint#196]
## Description

zsh-lint flags a valid zsh-only parameter expansion as a "bash feature", e.g.:

```
zsh/zsh/.zshrc:141:31: this expansion operator is a bash feature; tried parsing as zsh
```

The flagged line (from the well-known zsh vi-surround binding snippet documented in `zshzle`/`zsh-users`):

```zsh
for sequence in {a,i}${(s..)^:-'()[]{}<>bB'}; do bindkey -M $keymap $sequence select-bracketed; done
```

## Minimal repro

```zsh
for x in {a,i}${(s..)^:-'ab'}; do echo "$x"; done
```

This is **not** valid bash at all — bash rejects it outright:

```
$ bash -c 'for x in {a,i}${(s..)^:-x}; do echo "$x"; done'
bash: line 1: a${(s..)^:-x}: bad substitution
```

But it runs correctly under zsh, producing `aa ab ai a( a) a[ a] a{ a} a< a> ab aB ia ib ii i( i) i[ i] i{ i} i< i> ib iB` (RC-expand-with-brace via `^`, split-by-empty-string via `(s..)`, default value via `:-`).

## Expected behavior

zsh-lint should recognize `${(s..)^:-...}` (and similar `(s..)` + `^` + `:-` combinations) as valid zsh parameter-expansion flags rather than classifying the construct as bash-only, especially when it appears nested inside a brace expansion like `{a,i}${...}`.

## Environment

- zsh-lint installed via zi/git, run as `zsh-lint <file>`
- Triggered on a real-world `.zshrc` using the standard zsh vi-mode surround bindings

**Comment by ss-o** [comment]
Thanks for the precise repro. The cause was in the rc-expand caret compatibility path: it only recognized `${^name}` and not a caret following a parameter flag group, so `${(s..)^:-...}` fell through to the bash-only `^` case-modification diagnostic.

#200 extends the check to `${(flags)^...}` and adds your `bindkey` line to the parser corpus. The fix lands on `main` with that merge and will be in the next tagged release.
